linkml/transformers/rollup_transformer.py

```diff
@@ -0,0 +1,133 @@
+        class_designator_slot: The name of the slot that designates the original class (e.g., 'category').
+        include_all_classes: If True, keeps all classes in the schema; otherwise, only the target class
+            and non-descendants are kept.
+        include_mixins: If True, includes properties from mixins in the flattened class.
+    """
+
+    preserve_class_designator: bool = True
+    class_designator_slot: Optional[str] = "category"
+    include_all_classes: bool = False
+    include_mixins: bool = True
+
+
+class RollupTransformer(ModelTransformer):
+    """
+    Transforms a LinkML schema by flattening an inheritance hierarchy into a single class.
+
+    This transformer takes a schema and a target class name, and produces a new schema where
+    all descendant classes of the target class are removed and their slots are rolled up into the
+    target class.
+    """
+
+    def __init__(
+        self,
+        target_class: str,
+        config: Optional[FlattenTransformerConfiguration] = None,
+    ):
+        super().__init__()
+        self.target_class = target_class
+        self.config = config or FlattenTransformerConfiguration()
+        self._descendant_classes: set[str] = set()
+        self._class_to_slot_map: dict[str, set[str]] = {}
+        self._collect_slots_for_class: dict[str, list[SlotDefinition]] = {}
+
+    def transform(self, tgt_schema_name: str = None, top_class: ClassDefinitionName = None) -> SchemaDefinition:
+        """
+        Transform the schema by flattening the inheritance hierarchy.
+
+        Args:
+            tgt_schema_name: Optional target schema name (not used in this transformer)
+            top_class: Optional top class name (not used in this transformer)
+
+        Returns:
+            A transformed SchemaDefinition with the target class having all slots from its descendants.
+        """
+        if not self.schemaview:
+            raise ValueError("SchemaView not set. Ensure schema is set by calling set_schema().")
+
+        schema_copy = copy.deepcopy(self.source_schema)
+        view = SchemaView(schema_copy)
+        descendants = view.class_descendants(self.target_class, imports=True, mixins=True, reflexive=False, is_a=True)
+        self._descendant_classes = set(descendants)
+        self._collect_slots(view)
+        return self._create_flattened_schema(schema_copy, view)
+
+    def _collect_slots(self, view: SchemaView) -> None:
+        """
+        Collect all slots from the target class and its descendant classes.
+        Populates a mapping of class name to the set of its slot names.
+        """
+        # Get induced slots for the target class.
+        target_slots = view.class_induced_slots(self.target_class)
+        self._collect_slots_for_class[self.target_class] = target_slots
+
+        for descendant in self._descendant_classes:
+            slots = view.class_induced_slots(descendant)
+            self._collect_slots_for_class[descendant] = slots
```

Comment:
This is a slightly confusing name as it sounds like a method. Can you call it something like `self._class_to_induced_slot_map`, and maybe have the other one called `self._class_to_induced_slot_name_map` (since it's the same slots, but just records the slot names)?

Reply:
On second thoughts, `self._collect_slots_for_class` doesn't get used once it is set, so there's no point in keeping it. Naming problem solved!